code/parachain/frame/farming/src/lib.rs

```diff
@@ -0,0 +1,390 @@
+        AssetIdOf<T>, // lp token
+        Blake2_128Concat,
+        AssetIdOf<T>, // reward currency
+        RewardScheduleOf<T>,
+        ValueQuery,
+    >;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    // The pallet's dispatchable functions.
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Create or overwrite the reward schedule, if a reward schedule
+        /// already exists for the rewards currency the duration is added
+        /// to the existing duration and the rewards per period are modified
+        /// s.t. that the total (old remaining + new) rewards are distributed
+        /// over the new total duration
+        #[pallet::call_index(0)]
+        #[pallet::weight(T::WeightInfo::update_reward_schedule())]
+        #[transactional]
+        pub fn update_reward_schedule(
+            origin: OriginFor<T>,
+            pool_currency_id: AssetIdOf<T>,
+            reward_currency_id: AssetIdOf<T>,
+            period_count: u32,
+            #[pallet::compact] amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+            // pool_currency_id.sort();
+
+            // fund the pool account from treasury
+            let treasury_account_id = T::TreasuryAccountId::get();
+            let pool_account_id = Self::pool_account_id(&pool_currency_id);
+            T::MultiCurrency::transfer(reward_currency_id, &treasury_account_id, &pool_account_id, amount)?;
+
+            RewardSchedules::<T>::try_mutate(pool_currency_id, reward_currency_id, |reward_schedule| {
+                let total_period_count = reward_schedule
+                    .period_count
+                    .checked_add(period_count)
+                    .ok_or(ArithmeticError::Overflow)?;
+                let total_free = T::MultiCurrency::free_balance(reward_currency_id, &pool_account_id);
+                let total_per_period = total_free.checked_div(&total_period_count.into()).unwrap_or_default();
+
+                reward_schedule.period_count = total_period_count;
+                reward_schedule.per_period = total_per_period;
+
+                Self::deposit_event(Event::RewardScheduleUpdated {
+                    pool_currency_id,
+                    reward_currency_id,
+                    period_count: total_period_count,
+                    per_period: total_per_period,
+                });
+                Ok(().into())
+            })
+        }
+
+        /// Explicitly remove a reward schedule and transfer any remaining
+        /// balance to the treasury
+        #[pallet::call_index(1)]
+        #[pallet::weight(T::WeightInfo::remove_reward_schedule())]
+        #[transactional]
+        pub fn remove_reward_schedule(
+            origin: OriginFor<T>,
+            pool_currency_id: AssetIdOf<T>,
+            reward_currency_id: AssetIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            // pool_currency_id.sort();
```

Comment:
why comment?

Reply:
good catch. I need to remove this comment.